serving/purgatorio/select-and-train-machine-learning-models/deep-learning-theory/deep-learning-theory.md

```diff
@@ -0,0 +1,324 @@
+ 
+## Generative Adversarial Networks
+
+Generative Adversarial Networks, are an approach to generative modeling using deep learning methods, such as convolutional neural networks.
+
+Generative modeling is an [unsupervised learning](https://en.wikipedia.org/wiki/Unsupervised_learning) task in machine learning that involves automatically discovering and learning the regularities or patterns in input data in such a way that the model can be used to generate or output new examples that plausibly could have been drawn from the original dataset.
+
+GANs are a smart way of training a generative model by framing the problem as a supervised learning problem with two sub-networks: the generator network that we train to generate new examples, and the discriminator network that tries to classify examples as either real (from the domain) or fake (generated). The two networks are trained together in a [zero-sum game](https://en.wikipedia.org/wiki/Zero-sum_game), "adversarial", until the discriminator model is fooled about half the time, meaning the generator model is generating plausible examples.
+
+_First look (in order):_
+- [Here](https://machinelearningmastery.com/what-are-generative-adversarial-networks-gans/) you find a gentle introduction to Generative Adversarial Networks.
+- [Play with a GAN!](https://poloclub.github.io/ganlab/)
+- [This](https://developers.google.com/machine-learning/gan) is a tutorial from Google which gets you started in training GANs in real-world scenarios.
+- [Here](https://machinelearningmastery.com/impressive-applications-of-generative-adversarial-networks/) you can read which are some example applications of Generative Adversarial Networks.
+- [Here](https://www.youtube.com/watch?v=3-qazNQS2JU&t=1629s) you can find a panoramic overview of GANs "applications explosion" from their inventor (Ian Goodfellow). He's also co-author of the [Deep Learning Book](http://www.deeplearningbook.org/)!
+- [Here](https://www.youtube.com/watch?v=KudkR-fFu_8) you find a detailed Theory and Applications video on GANs.
+
+_Dive Deeper_: Use the latest "survey paper" to choose which papers are worth exploring for your case and have a nice panoramic of the state of the art:
+[Generative Adversarial Networks: recent developments](https://arxiv.org/abs/1903.12266)
+
+_Second pass_: [Generative Models Chapter](http://www.deeplearningbook.org/contents/generative_models.html).
+
+## AutoEncoders 
+
+Autoencoders are a particular class of Deep Learning algorithms that tries to, given and input, compress it in the internal layers and then reconstruct it as approaching the output layer, with the final output of the network that should be similar as much as possible to the input. They can be used to a variety of tasks, for example, denoising data, compress the data to visualize them, and use some trained layers of an Autoencoder to bootstrap the learning process of another Neural Network. 
```

Comment:
Footnote: based on the same principles of compression/decompression algorithms (zip/unzip archives)